Use backup providers

dveeden opened the pull request:
Add backup providers for
* mysqldump
* MySQL Enterprise Backup
* Percona XtraBackup

dveeden commented:
This is an alternative for #647

igalic commented:
i'm kinda surprised this is still passing spec tests, for one, for the other, as you can imagine, i prefer this solution, and would as such prefer it to be (better?) documented

hunner commented on manifests/server/backup.pp:
```diff
@@ -1,62 +1,43 @@
-    owner  => $backupdirowner,
-    group  => $backupdirgroup,
-  }
+  create_resources("mysql::backup::${provider}", {
```
Could you convert the backup defines to classes (since they will only ever be used once) and have this create_resources of type "class")?

dveeden replied:
@hunner done. Looks better.

hunner commented:
Descriptions in the readme about the different providers would be great, and document the limitations/requirements (eg: the `meb` package needs to exist in a repo).

hunner commented on manifests/backup/mysqlbackup.pp:
```diff
@@ -0,0 +1,85 @@
+# See README.me for usage.
+define mysql::backup::mysqlbackup (
```
Should this be called mysqlenterprisebackup instead?

dveeden replied:
No, I don't think so. The full name is MySQL Enterprise Backup and the
package name is meb and the tool/binary which you actually need to call
is mysqlbackup.

On 02/05/2015 07:36 PM, Hunter Haugen wrote:
>
> In manifests/backup/mysqlbackup.pp
> <https://github.com/puppetlabs/puppetlabs-mysql/pull/649#discussion_r24186509>:
>
> > @@ -0,0 +1,85 @@
> > +# See README.me for usage.
> > +define mysql::backup::mysqlbackup (
>
> Should this be called mysqlenterprisebackup instead?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppetlabs-mysql/pull/649/files#r24186509>.
>

dveeden commented:
I've added a short description to the README.md and I've added a config file for mysqlbackup (separate from the main my.cnf so it can be protected with 0600).

igalic commented:
thaaaaanks @dveeden!
and thank you @hunner for the review!
